### PR TITLE
Assorted bug fixes

### PR DESCRIPTION
**Pull request by careo**
Including:
 * Fix bug in Hash with [] for keys not in example hashes.
 * Add some file content to example Attributor::Tempfiles.

Signed-off-by: Dane Jensen <dane@rightscale.com>

### CODE REVIEW

File: lib/attributor/types/hash.rb [hunk 1]
```diff
@@ -288,6 +288,24 @@ def generate_subcontext(context, key_name)
     def get(key, context: self.generate_subcontext(Attributor::DEFAULT_ROOT_CONTEXT,key))
       key = self.class.key_attribute.load(key, context)
 
+      if self.class.keys.empty?
```
Comment (blanquer) [inline lib/attributor/types/hash.rb]: I forget what this fixed exactly...maybe we should add it to the changelog...

File: lib/attributor/types/tempfile.rb [hunk 2]
```diff
@@ -9,10 +9,15 @@ def self.native_type
...
-      ::Tempfile.new(Attributor.humanize_context(context))
+      file = ::Tempfile.new(Attributor.humanize_context(context))
+      file.write /[:sentence:]/.gen
+      file.write '.'
+      file.rewind
+      file
     end
 
     def self.dump(value, **opts)
+      #value && value.read
```
Comment (blanquer) [inline lib/attributor/types/tempfile.rb]: don't we want to dump the actual file contents?

File: lib/attributor/types/hash.rb (hunk 1 above)
Comment (careo) [inline lib/attributor/types/hash.rb]: It fixed `get` on a hash with no defined keys breaking.

Comment (careo) [inline lib/attributor/types/hash.rb]: Updated changelog with this.

File: lib/attributor/types/tempfile.rb (hunk 2 above)
Comment (careo) [inline lib/attributor/types/tempfile.rb]: Oh true, I suppose we do now. Fixed.

**Comment by blanquer** [comment]
LGTM

**Pull request merged**